### PR TITLE
FMWK-150 Deeply nested POJO/field

### PR DESCRIPTION
**Pull request by agrgr**

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 1]
```diff
@@ -847,6 +886,14 @@ public Filter sIndexFilter(Map<String, Object> map) {
         }
     };
 
+    private static CTX[] dotPathToCtx(String[] dotPathArray) {
+        List<CTX> list = Arrays.stream(dotPathArray).map(str -> CTX.mapKey(Value.get(str)))
+            .collect(Collectors.toList());
+        list.remove(0); // first element is bin name
+        list.remove(list.size() - 1); // last element is the key we already have
+        return list.toArray(CTX[]::new);
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:894]:
```suggestion
        return Arrays.stream(dotPathArray).map(str -> CTX.mapKey(Value.get(str)))
            .skip(1) // first element is bin name
            .limit(dotPathArray.length - 2L) // last element is the key we already have
            .toArray(CTX[]::new);
```

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 2]
```diff
@@ -279,22 +283,58 @@ public Filter sIndexFilter(Map<String, Object> map) {
...
         @Override
         public Exp filterExp(Map<String, Object> map) {
-            // VALUE2 contains key (field name)
+            String dotPath = getDotPath(map);
+            String[] dotPathArr;
+            if (StringUtils.hasLength(dotPath)) {
+                dotPathArr = dotPath.split("\\.");
+            } else {
+                throw new RuntimeException("MAP_VALUE_EQ_BY_KEY: dotPath has not been set");
+            }
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:292]: Extract to a method.

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 3]
```diff
@@ -279,22 +283,58 @@ public Filter sIndexFilter(Map<String, Object> map) {
...
+
+            boolean useCtx = dotPathArr.length > 2;
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:294]: final

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 4]
```diff
@@ -279,22 +283,58 @@ public Filter sIndexFilter(Map<String, Object> map) {
     MAP_VALUE_EQ_BY_KEY {
         @Override
         public Exp filterExp(Map<String, Object> map) {
-            // VALUE2 contains key (field name)
+            String dotPath = getDotPath(map);
+            String[] dotPathArr;
+            if (StringUtils.hasLength(dotPath)) {
+                dotPathArr = dotPath.split("\\.");
+            } else {
+                throw new RuntimeException("MAP_VALUE_EQ_BY_KEY: dotPath has not been set");
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:291]: Use `IllegalArgumentException`

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 5]
```diff
@@ -307,18 +347,38 @@ public Exp filterExp(Map<String, Object> map) {
          */
         @Override
         public Filter sIndexFilter(Map<String, Object> map) {
+            String dotPath = getDotPath(map);
+            String[] dotPathArr;
+            if (StringUtils.hasLength(dotPath)) {
+                dotPathArr = dotPath.split("\\.");
+            } else {
+                throw new RuntimeException("MAP_VALUE_EQ_BY_KEY: dotPath has not been set");
+            }
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:356]: Reuse the extracted method.

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:1]: Optimize imports.

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java (hunk 4 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:291]: There is no argument that can be passed though. It is done in QueryCreator and I put a RuntimeException here to signalize that the flow is incorrect.

**Review by reugn — APPROVED** [review]

**Pull request merged**